File: src/main/java/com/github/njuro/jard/config/security/jwt/JwtAuthenticationEntryPoint.java
package com.github.njuro.jard.config.security.jwt;

import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.web.AuthenticationEntryPoint;
import org.springframework.stereotype.Component;

/** Handles AuthenticationException on failed JWT authentication. */
@Component
public class JwtAuthenticationEntryPoint implements AuthenticationEntryPoint {

  @Override
  public void commence(
      HttpServletRequest httpServletRequest,
      HttpServletResponse httpServletResponse,
      AuthenticationException e)
      throws IOException, ServletException {
    httpServletResponse.sendError(HttpServletResponse.SC_UNAUTHORIZED, e.getMessage());
  }
}

File: src/main/kotlin/com/github/njuro/jard/rewrite/config/security/jwt/JwtAuthenticationEntryPoint.kt
package com.github.njuro.jard.rewrite.config.security.jwt

import org.springframework.security.core.AuthenticationException
import org.springframework.security.web.AuthenticationEntryPoint
import org.springframework.stereotype.Component
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse
import javax.servlet.http.HttpServletResponse.SC_UNAUTHORIZED

/** Handles AuthenticationException on failed JWT authentication.  */
@Component
class JwtAuthenticationEntryPoint : AuthenticationEntryPoint {
    override fun commence(
        httpServletRequest: HttpServletRequest,
        httpServletResponse: HttpServletResponse,
        ex: AuthenticationException
    ) {
        httpServletResponse.sendError(SC_UNAUTHORIZED, ex.message)
    }
}
